packages/editor/src/store/utils/notice-builder.js

```diff
@@ -95,13 +95,17 @@ export function getNotificationArgumentsForSaveFail( data ) {
 	// If the post was being published, we show the corresponding publish error message
 	// Unless we publish an "updating failed" message
 	const messages = {
-		publish: __( 'Publishing failed' ),
-		private: __( 'Publishing failed' ),
-		future: __( 'Scheduling failed' ),
+		publish: __( 'Publishing failed.' ),
+		private: __( 'Publishing failed.' ),
+		future: __( 'Scheduling failed.' ),
 	};
-	const noticeMessage = ! isPublished && publishStatus.indexOf( edits.status ) !== -1 ?
+	let noticeMessage = ! isPublished && publishStatus.indexOf( edits.status ) !== -1 ?
 		messages[ edits.status ] :
-		__( 'Updating failed' );
+		__( 'Updating failed.' );
+
+	if ( ! ( /<\/?[^>]*>/.test( error.message ) ) ) {
```

Comment:
Is the idea with this to exclude messages with HTML? I think it may be agreeable as a first-pass improvement, but is there a fundamental reason we should want to exclude messages containing HTML?

Stripping the HTML tags seems reasonable to me, though the specifics of doing so are some combination of [challenging](https://unpkg.com/striptags@3.1.1/src/striptags.js) and [risky](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.md#rule-1---html-escape-then-javascript-escape-before-inserting-untrusted-data-into-html-subcontext-within-the-execution-context) ([example solution](https://stackoverflow.com/a/822486)), depending if we consider the REST response to be "safe".

It might also help to provide some context to the future maintainer what's actually happening here, either with an inline code comment or extracting as a separate, helpfully-named function (e.g. `stripTags`), in scope or proposed as part of a relevant package (closest might be [`escape-html`](https://github.com/WordPress/gutenberg/tree/master/packages/escape-html)).

Reply:
There reason we don't want html tags here, as some responses may contain hyperlinks or bullet points. The content may not make sense anymore if it says for example "Click on this link" and it is no longer a link. 

I didn't make the tag check into a function, as I didn't seem like it would be useful to do so. But I can, if you so wish.

Reply:
> There reason we don't want html tags here, as some responses may contain hyperlinks or bullet points. The content may not make sense anymore if it says for example "Click on this link" and it is no longer a link.

I think it's okay to be conservative here as you propose, as long as there's the fallback messaging of "Publishing failed".

> I didn't make the tag check into a function, as I didn't seem like it would be useful to do so. But I can, if you so wish.

My concern here is whether it will be obvious here to a future maintainer what this logic is intended to do. Whether that's an inline code comment or function, I have no strong preference. But as implemented currently, I foresee it being not immediately obvious to its purpose.

Reply:
>My concern here is whether it will be obvious here to a future maintainer what this logic is intended to do. Whether that's an inline code comment or function, I have no strong preference. But as implemented currently, I foresee it being not immediately obvious to its purpose.

Can we add a code comment here?